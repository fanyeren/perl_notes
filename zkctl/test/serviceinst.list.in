#!/bin/bash
source $(dirname $0)/common

TYPE=service-instance
NODE=hankjin.vicp.net

node_clear $TYPE
node_clear service
./zkctl -s slavenode add --slavenode-id=$NODE  --data={}

./zkctl -w $TYPE add --service-id=bs_dnews_0 --slavenode-id=$NODE --data={} 2>&1
echo "Add with not exist service  failed: result=$?"
./zkctl -w service add --service-id=bs_dnews_0  --data={}
./zkctl -w service add --service-id=bs_dnews_2  --data={}
./zkctl -w service add --service-id=bs_wp_2     --data={}
echo "add service succedd result=$?"
./zkctl -w service add --service-id=bc_group_1 --data={}
#正常添加
id=$(./zkctl -w $TYPE add --service-id=bs_dnews_0 --slavenode-id=$NODE --data={})
echo "Add service instance $id succeed: result=$?"
id=$(./zkctl -w $TYPE add --service-id=bs_dnews_2 --slavenode-id=$NODE --data={})
echo "Add service instance $id succeed: result=$?"
id=$(./zkctl -w $TYPE add --service-id=bs_wp_2 --slavenode-id=$NODE --data={})
echo "Add service instance $id succeed: result=$?"
bcid=$(./zkctl -w $TYPE add --service-id=bc_group_1 --slavenode-id=$NODE --data={})
echo "Add service instance $bcid succeed: result=$?"
./zkctl -s $TYPE list | sort

#删除service
./zkctl -w service del --service-id=bc_group_1 2>&1
echo "delete service should failed. result=$?"
#删除
./zkctl -w $TYPE del --$TYPE-id=$bcid
echo "delete bc instance $bcid result=$?"
./zkctl -w $TYPE list | sort

./zkctl -w service del --service-id=bc_group_1 2>&1
echo "delete service should succeed. result=$?"
#删除已经删除的
./zkctl -w $TYPE del --$TYPE-id=$bcid 2>&1
echo "Delete already deleted result=$?"

node_clear $TYPE
node_clear service
node_clear slavenode
