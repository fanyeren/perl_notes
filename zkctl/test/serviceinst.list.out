service-instance add failed, can't find service "bs_dnews_0"
Add with not exist service  failed: result=1
add service succedd result=0
Add service instance bs_dnews_0-0000000001 succeed: result=0
Add service instance bs_dnews_2-0000000001 succeed: result=0
Add service instance bs_wp_2-0000000001 succeed: result=0
Add service instance bc_group_1-0000000001 succeed: result=0
bc_group_1-0000000001
bs_dnews_0-0000000001
bs_dnews_2-0000000001
bs_wp_2-0000000001
service del failed, probally has child
delete service should failed. result=1
delete bc instance bc_group_1-0000000001 result=0
bs_dnews_0-0000000001
bs_dnews_2-0000000001
bs_wp_2-0000000001
delete service should succeed. result=0
service-instance bc_group_1-0000000001 del failed, can not find this
Delete already deleted result=1
